Spook/Oracles/CoinGecko.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using LunarLabs.Parser.JSON;
+using System.Net.Http;
+
+namespace Phantasma.Spook.Oracles
+{
+    public static class CoinGeckoUtils
+    {
+        public static decimal GetCoinRate(string baseSymbol, string quoteSymbol, PricerSupportedToken[] supportedTokens)
+        {
+
+            string json;
+            string baseticker = "";
+
+            foreach (var token in supportedTokens)
+            {
+                if(token.ticker == baseSymbol)
+                {
+                    baseticker = token.coingeckoId;
+                    break;
+                }
+            }
+
+            if (String.IsNullOrEmpty(baseticker))
+                return 0;
+
+            var url = $"https://api.coingecko.com/api/v3/simple/price?ids={baseticker}&vs_currencies={quoteSymbol}";
+
+            try
+            {
+
+                using (var httpClient = new HttpClient())
```

Comment:
I think Webclient is better suited, it's more lightweight so costs less to setup if a new object is created each time called.